src/dbc.rs

```diff
@@ -186,142 +403,120 @@ mod tests {
 
-        let mut dbc_builder = DbcBuilder::new(reissue_request.transaction);
+        let mut dbc_builder = DbcBuilder::new(revealed_commitments, output_owners);
         dbc_builder = dbc_builder.add_reissue_share(split_reissue_share);
         let output_dbcs = dbc_builder.build()?;
 
         // The outputs become inputs for next reissue.
-        let inputs = output_dbcs.into_iter().map(|dbc| {
-            let amount_secrets =
-                DbcHelper::decrypt_amount_secrets(&input_owner, &dbc.content).unwrap();
-            (dbc, amount_secrets)
-        });
-
-        let reissue_tx = crate::TransactionBuilder::default()
-            .add_inputs(inputs)
-            .add_output(crate::Output {
-                amount,
-                owner: crate::bls_dkg_id().public_key_set.public_key(),
+        let inputs: Vec<(Dbc, SecretKey, Vec<DecoyInput>)> = output_dbcs
+            .into_iter()
+            .map(|(dbc, owner_once, _amount_secrets)| {
+                (
+                    dbc,
+                    owner_once.owner_base().secret_key().unwrap(),
+                    spentbook.random_decoys(STD_NUM_DECOYS, &mut rng8),
+                )
             })
-            .build()?;
-
-        let tx_blinded = reissue_tx.blinded();
+            .collect();
+
+        let owner_once =
+            OwnerOnce::from_owner_base(Owner::from_random_secret_key(&mut rng), &mut rng8);
+
+        let (reissue_tx, _revealed_commitments, material, _output_owners) =
+            crate::TransactionBuilder::default()
+                .add_inputs_dbc(inputs, &mut rng8)?
+                .add_output(
+                    Output {
+                        amount,
+                        public_key: owner_once.as_owner().public_key_blst(),
+                    },
+                    owner_once.clone(),
+                )
+                .build(&mut rng8)?;
 
         let mut rr_builder = ReissueRequestBuilder::new(reissue_tx.clone());
-        for input in reissue_tx.inputs.iter() {
-            let spent_sig = input_owner.public_key_set.combine_signatures([(
-                input_owner.index,
-                input_owner
-                    .secret_key_share
-                    .derive_child(&input.spend_key_index())
-                    .sign(tx_blinded.hash()),
-            )])?;
-            let spentbook_pks = genesis_node.key_manager.public_key_set()?;
-            let spentbook_sig_share = genesis_node
-                .key_manager
-                .sign(&SpentProof::proof_msg(&tx_blinded.hash(), &spent_sig))?;
-
-            rr_builder = rr_builder.add_spent_proof_share(
-                input.spend_key(),
-                SpentProofShare {
-                    spent_sig,
-                    spentbook_pks,
-                    spentbook_sig_share,
-                },
-            );
+        for (i, (mlsag, _mlsag_material)) in reissue_tx
+            .mlsags
+            .iter()
+            .zip(material.inputs.iter())
```

Comment:
Why zip if we are not using the `mlsag_material`?

Reply:
fixed.